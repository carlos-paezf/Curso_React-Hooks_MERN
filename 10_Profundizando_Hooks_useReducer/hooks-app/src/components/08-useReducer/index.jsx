export { TodoApp } from "./TodoApp";
export { TodoList } from "./TodoList";
export { TodoListItem } from "./TodoListItem";
export { TodoAdd } from "./TodoAdd";