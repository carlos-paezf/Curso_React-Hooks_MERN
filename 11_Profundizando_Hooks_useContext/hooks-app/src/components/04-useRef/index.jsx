export { FocusScreen } from './FocusScreen'
export { RealExampleRef } from './RealExampleRef'