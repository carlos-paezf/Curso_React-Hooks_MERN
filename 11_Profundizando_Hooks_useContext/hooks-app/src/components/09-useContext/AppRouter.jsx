import React from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom'
import { AboutScreen, HomeScreen, LoginScreen, NavBar } from '.';


export const AppRouter = () => {
    return (
        <Router>
            <NavBar />
            <div className="container mt-3">
                <Switch>
                    <Route exact path="/" component={HomeScreen} />
                    <Route exact path="/login" component={LoginScreen} />
                    <Route exact path="/about" component={AboutScreen} />
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    );
}
