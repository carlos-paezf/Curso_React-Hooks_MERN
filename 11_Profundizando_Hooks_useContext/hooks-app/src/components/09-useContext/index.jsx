export { AppRouter } from "./AppRouter";

export { AboutScreen } from "./AboutScreen";
export { HomeScreen } from "./HomeScreen";
export { LoginScreen } from "./LoginScreen";
export { MainApp } from "./MainApp";

export { NavBar } from "./NavBar";