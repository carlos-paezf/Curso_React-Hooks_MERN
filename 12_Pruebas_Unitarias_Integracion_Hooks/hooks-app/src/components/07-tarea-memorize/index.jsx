export { Hijo } from "./Hijo";
export { Padre } from "./Padre";