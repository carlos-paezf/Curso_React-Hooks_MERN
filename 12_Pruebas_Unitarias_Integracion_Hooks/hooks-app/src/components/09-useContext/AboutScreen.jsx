import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const AboutScreen = () => {

    const {user, setUser} = useContext(UserContext)

    return (
        <>
            <h2>About Screen</h2>
            <hr />
            <pre className='container'>{ JSON.stringify(user, null, 5) }</pre>
            <button className="btn btn-danger" onClick={() => setUser({})}>Logout</button>
        </>
    )
}
