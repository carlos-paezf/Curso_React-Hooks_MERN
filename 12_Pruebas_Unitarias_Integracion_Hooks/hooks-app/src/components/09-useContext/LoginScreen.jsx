import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const { user, setUser } = useContext(UserContext)

    const handleLogin = () => {
        setUser({ 
            id: 1234, 
            name: 'ferrer',
            email: 'test@mail.com'
        })
    };


    return (
        <>
            <h2>Login Screen</h2>
            <hr />
            <pre className='container'>{ JSON.stringify(user, null, 5) }</pre>
            <button className="btn btn-primary" onClick={handleLogin}>Login</button>
        </>
    )
}