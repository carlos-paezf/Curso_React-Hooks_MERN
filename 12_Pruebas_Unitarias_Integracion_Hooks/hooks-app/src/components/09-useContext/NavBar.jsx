import React from 'react';
import { NavLink } from 'react-router-dom'


export const NavBar = () => {

    const routes = [
        { text: 'Home', path: '/' },
        { text: 'About', path: '/about' },
        { text: 'Login', path: '/login' }
    ]

    return (
        <nav className='navbar navbar-expand-lg navbar-dark bg-dark'>
            <div className="container-fluid">
                <ul className='navbar-nav me-auto mb-2 mb-lg-0'>
                    {
                        routes.map(({ path, text }, i) => (
                            <li key={i} className='nav-item'>
                                <NavLink exact to={path} className="nav-link" activeClassName='active'>{text}</NavLink>
                            </li>
                        ))
                    }
                </ul>
            </div>
        </nav>
    )
}
