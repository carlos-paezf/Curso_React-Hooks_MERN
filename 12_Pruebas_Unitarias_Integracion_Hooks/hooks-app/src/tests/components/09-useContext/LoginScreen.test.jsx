import React from 'react';
import { mount } from 'enzyme'
import { LoginScreen } from '../../../components/09-useContext';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas al componente <LoginScreen />', () => {
    let wrapper

    const setUser = jest.fn()

    beforeEach(() => {
        wrapper = mount(<UserContext.Provider value={{ setUser }}>
            <LoginScreen />
        </UserContext.Provider>)
    })

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe hacer login con los argumentos especificos', () => {
        wrapper.find('button').simulate('click');

        expect(setUser).toHaveBeenCalledTimes(1)
        expect(setUser).toHaveBeenCalledWith({
            id: 1234,
            name: 'ferrer',
            email: 'test@mail.com'
        })
    })
})