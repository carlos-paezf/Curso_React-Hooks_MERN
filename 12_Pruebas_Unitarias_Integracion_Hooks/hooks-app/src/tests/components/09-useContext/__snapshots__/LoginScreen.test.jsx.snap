// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <LoginScreen /> Debe mostrarse correctamente 1`] = `
Array [
  <h2>
    Login Screen
  </h2>,
  <hr />,
  <pre
    className="container"
  />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
